xbmc/cores/VideoPlayer/VideoRenderers/RenderManager.cpp

```diff
@@ -1083,9 +1085,7 @@ void CRenderManager::PrepareNextRender()
   double frameOnScreen = m_dvdClock.GetClock();
   double frametime = 1.0 / g_graphicsContext.GetFPS() * DVD_TIME_BASE;
 
-  // correct display latency
-  // internal buffers of driver, assume that driver lets us go one frame in advance
-  double totalLatency = DVD_SEC_TO_TIME(m_displayLatency) - DVD_MSEC_TO_TIME(m_videoDelay) + 2* frametime;
```

Comment:
Mh what do we want to do here then. It seems CGraphicContext is the only one to (more or less...) reliably remember the FPS which is needed for the default calculation. But I can't put the Wayland stuff in there obviously. How about this: I remove the default calculation with the buffer number from `CWinSystemBase` and have it return some kind of invalid value by default. `CRenderManager` can then do the default calculation itself as fallback and ask `CGraphicContext` for the FPS.